src/Language/PureScript/Make.hs

```diff
@@ -130,34 +131,64 @@ rebuildModuleWithIndex MakeActions{..} exEnv externs m@(Module _ _ moduleName _
                Right d -> d
 
   evalSupplyT nextVar'' $ codegen renamed docs exts
+  -- evaluate $ trace ("\n===== externs: " <> show moduleName <> ":\n" <> show exts) ()
   return exts
 
--- | Compiles in "make" mode, compiling each module separately to a @.js@ file and an @externs.cbor@ file.
+-- | Compiles in "make" mode, compiling each module separately to a @.js@ file
+-- and an @externs.cbor@ file.
 --
--- If timestamps or hashes have not changed, existing externs files can be used to provide upstream modules' types without
--- having to typecheck those modules again.
+-- If timestamps or hashes have not changed, existing externs files can be used
+-- to provide upstream modules' types without having to typecheck those modules
+-- again.
+--
+-- This version will collect an return externs only of modules that were used
+-- during the build.
 make :: forall m. (MonadBaseControl IO m, MonadError MultipleErrors m, MonadWriter MultipleErrors m)
      => MakeActions m
      -> [CST.PartialResult Module]
      -> m [ExternsFile]
-make ma@MakeActions{..} ms = do
+make ma ms = makeImp ma ms False
+
+-- | Compiles in "make" mode, compiling each module separately to a @.js@ file
+-- and an @externs.cbor@ file.
+--
+-- If timestamps or hashes have not changed, existing externs files can be used
+-- to provide upstream modules' types without having to typecheck those modules
+-- again.
+--
+-- This version will collect an return all externs of all passed modules.
+make' :: forall m. (MonadBaseControl IO m, MonadError MultipleErrors m, MonadWriter MultipleErrors m)
```

Comment:
Doesn't this case correspond to the original behavior of `make`? If so, this function should be called `make` and the other one `makeAndReturnUsed` (or something); I don't think the `'` is clear enough in this case to distinguish the two behaviors, particularly if the `'` is the old behavior and the bare `make` is the new one.

I would also be fine with having `makeImp` be the new `make` and using one or the other of the constructors of `ExternsFileCollectionPolicy` (per Filip's suggestion; or whatever name you end up using) at every call site, instead of going through these two auxiliary functions.